lib/app-routes/common/auth/keys.js

```diff
@@ -8,6 +8,8 @@ const _ = require('lodash');
 
 const baseUrl = '/keys';
 
+const  recordObj = {};
```

Comment:
By declaring `recordObj` up here, you have created a **race condition**. This is because `recordObj` is mutated within the route handler, and it is possible for the route handler to be called multiple times asynchronously as login requests are received. 

You can fix this by moving the declaration of `recordObj` into the route handler itself, then modifying `saveLoginRecord()` to accept a single parameter, `properties`, which will be used to forge the LoginRecord.